clients/python/src/model_registry/integrator.py

```diff
@@ -0,0 +1,180 @@
+                return mlflowSourceUri
+
+        for pattern in regex_patterns:
+            if re.match(pattern, mlflowSourceUri):
+                return mlflowSourceUri
+
+        warn("Unsupported artifact uri format, may not accessible by downstream application", stacklevel=2)
+        return mlflowSourceUri
+
+    def _getMlflowModelInfo(self, tracking_uri: str, registered_name: str, registered_version: str, model_name: str, author: str = None) -> dict:
+        try:
+            import mlflow
+            from mlflow.exceptions import RestException
+        except ImportError as e:
+            msg = "mlflow is not installed"
+            raise StoreException(msg) from e
+        mlflow.set_tracking_uri(tracking_uri)
+        client = mlflow.tracking.MlflowClient()
+        try:
+            if registered_version:
+                model_version_details = client.get_model_version(name=registered_name, version=registered_version)
+            else:
+                model_version_details = client.get_registered_model(registered_name).latest_versions[0]
+        except RestException as e:
+            msg = f"Mlflow API error: {e.message}"
+            raise StoreException(msg)
+
+        if not author:
+            if "author" in model_version_details.tags:
+                self._modelInfo["author"] = model_version_details.tags["author"]
+            else:
+                self._modelInfo["author"] = "unknown"
+        else:
+            self._modelInfo["author"] = author
+
+        sourceUri = self._validateArtifactUri(model_version_details.source)
+        model_file_name = mlflow.pyfunc.load_model(sourceUri).metadata.flavors["python_function"]["data"]
+        self._modelInfo.update(
+            {
+                "uri": f"{sourceUri}/{model_file_name}",
+                "metadata": {
+                    "name_in_origin_mr": model_version_details.name,
+                    "version_in_origin_mr": model_version_details.version
+                    "tracking_uri": tracking_uri,
+                    "model_origin": "MlFlow",
+                    "model_author": self._modelInfo["author"],
+                },
+            }
+        )
+
+        self._modelInfo.update(model_version_details.tags)
+        self._modelInfo["name"] = model_name or model_version_details.name
+        return self._modelInfo
+
+
+class ModelInfoManager:
+
+    @classmethod
+    def providers(cls):
+        return {
+            supportedPlatform.HUGGING_FACE: HuggingFaceModelInfoProvider(),
+            supportedPlatform.MLFLOW: MLflowModelInfoProvider(),
+        }
+
+    @classmethod
+    def get_model_info(cls, platform: str, kwargs: dict) -> dict:
+        assert platform in [p.value for p in supportedPlatform]
+        provider = cls.providers().get(supportedPlatform(platform))
+        if provider:
+            return provider.get_model_info(**kwargs)
+        else:
+            return {"error": "Platform not supported"}
```

Comment:
If we're providing custom methods for each provider, there's no need for any "Manager" boilerplate.